core/services/offchainreporting/peer_wrapper.go

```diff
@@ -97,6 +97,16 @@ func (p *SingletonPeerWrapper) Start() error {
 		var matched bool
 		checkedKeys := []string{}
 		configuredPeerID, err := p.config.P2PPeerID()
+
+		if configuredPeerID == "" {
+			if len(p2pkeys) == 1 {
+				logger.Warn("No P2P_PEER_ID set, defaulting to first key in database")
+				configuredPeerID = p2pkeys[0].PeerID()
```

Comment:
Setting this here makes the for loop below redundant. Can we clean up the logic?